vscode-dotnet-runtime-library/src/Utils/WebRequestWorker.ts

```diff
@@ -73,12 +77,50 @@ export class WebRequestWorker
             }
         });
 
-            this.client = setupCache(uncachedAxiosClient,
-                {
-                    storage: buildMemoryStorage(),
-                    ttl: this.cacheTimeToLive
-                }
-            );
+        // Record when the web requests begin:
+        // Register this so it happens before the cache https://axios-cache-interceptor.js.org/guide/interceptors#explanation
+        uncachedAxiosClient.interceptors.request.use( req => {
+            req.headers.startTime = process.hrtime.bigint();
+            return req;
+        });
+
+        // Record when the server responds:
+        // Register this so it happens after the cache -- this means we need to check if the result is cached before reporting perf data!
+        // ^ the request should not be processed if it is in the cache, it seems nonsensical to check this before the cache is hit even though this is possible
+        uncachedAxiosClient.interceptors.response.use(res =>
+        {
+            // res.request headers is undefined. Need to access it via the member. Could not find any documentation or other variable access to do this.
+
+            // _header has all of the headers, separated by \r\n, with kv pair Name: Content
+            // Split it into arrays of headers, trim the whitespace and empty headers, then filter for the startTime header only, then grab the only result (startTime header array) and its value.
+            // Could be done more efficiently.
+            let startTime;
+            try
+            {
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                startTime = BigInt(res.request._header?.split('\r\n')?.filter((headerLine : string) => headerLine !== '')?.map(
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                    (headerColonSpaceValue : any) => headerColonSpaceValue.split(':').map(
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                    (value : any) => value.trim())
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                ).filter((headerAndValue : any) => headerAndValue[0] === 'startTime')[0][1]);
+            }
+            catch(e : any)
+            {
+                // protect against being intercepted and having a weird header type
+            }
+            res.headers.startTime = startTime;
```

Comment:
Where is this actually used? Is it just added to a database somewhere?

Reply:
This is added in the web request and fetched later. But I think we dont want to keep that.